provider/datagen/src/transform/icuexport/collator/mod.rs

```diff
@@ -36,24 +36,63 @@ impl DataProvider<CollationFallbackSupplementV1Marker> for DatagenProvider {
     ) -> Result<DataResponse<CollationFallbackSupplementV1Marker>, DataError> {
         self.check_req::<CollationFallbackSupplementV1Marker>(req)?;
 
-        let data = LocaleFallbackSupplementV1 {
-            parents: self
-                .cldr()?
-                .core()
-                .read_and_parse::<crate::provider::transform::cldr::cldr_serde::parent_locales::Resource>(
-                    "supplemental/parentLocales.json",
-                )?
-                .supplemental
-                .parent_locales
-                .collations
-                .iter()
-                .map(|(from, to)| {
-                    (
-                        <&UnvalidatedStr>::from(from.as_str()),
-                        <(Language, Option<Script>, Option<Region>)>::from(to),
+        let parent_locales = &self
+            .cldr()?
+            .core()
+            .read_and_parse::<crate::provider::transform::cldr::cldr_serde::parent_locales::Resource>(
+                "supplemental/parentLocales.json",
+            )?.supplemental.parent_locales;
+
+        let additional = if parent_locales
+            .rules
+            .collations
+            .as_ref()
+            .map(|c| &c.non_likely_scripts)
+            != Some(&String::from("root"))
+        {
+            let collation_locales = self
+                .icuexport()?
+                .list(&format!("collation/{}", self.collation_han_database()))?
+                .filter_map(|s| {
+                    Some(
+                        file_name_to_locale(
+                            s.rsplit_once('_')?.0,
+                            self.has_legacy_swedish_variants(),
+                        )?
+                        .id
+                        .language,
                     )
                 })
-                .collect(),
+                .collect::<HashSet<_>>();
+
+            parent_locales
+                .parent_locale
```

Comment:
Suggestion: This only works so long as the non-likely scripts are still in the parent list, right? They are planned to be removed. A better place to source these might be `coverageLevels`.

Reply:
Huh? Where will the fallback `zh-Hant` -> `und` live?

Reply:
You need to check the likely subtags / construct a LocaleFallbacker.

But as discussed, this is correct for CLDR 45, so let's merge.